tests/app/notifications/test_receive_notification.py

```diff
@@ -166,7 +166,37 @@ def test_create_inbound_sms_object(
     assert inbound_sms.content == 'hello there 📩'
     assert inbound_sms.provider == 'twilio'
 
-    # Teardown
+
+def test_create_inbound_sms_object_logs_invalid_from_number(
+    notify_api,
+    mocker,
+    sample_service,
+):
+    service = sample_service()
+    ref = str(uuid4())
+    number = '+16502532222'
+    invalid_from_number = 'ALPHANUM3R1C'
+
+    mock_logger = mocker.patch('notifications_utils.recipients.logging.exception')
+
+    inbound_sms = create_inbound_sms_object(
+        service=service,
+        content='no matter where you go, there you are',
+        notify_number=number,
+        from_number=invalid_from_number,
+        provider_ref=ref,
+        date_received=datetime.utcnow(),
+        provider_name='twilio',
```

Comment:
Please import `TWILIO_PROVIDER`, we want to avoid hard coding any "email", "sms", "pinpoint", "twilio", etc.